src/clean-core/alloc_function.hh

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <clean-core/allocator.hh>
+#include <clean-core/always_false.hh>
+#include <clean-core/forward.hh>
+
+namespace cc
+{
+// std::function replacement, not copyable, with allocator support
+// slightly faster than unique_function, same codegen as STL (minus exceptions):
+// https://godbolt.org/z/8fY3a5
+// https://quick-bench.com/q/sCMOpZNIacJcwOPcCYt0_95Efoc
+template <class Signature>
+struct alloc_function;
+template <class Result, class... Args>
+struct alloc_function<Result(Args...)>
```

Comment:
move-only would mean alloc_unique_function for consistency if we want to have copyable alloc_function later...